src/EditorFeatures/Core/Implementation/IntelliSense/SignatureHelp/Controller.Session_UpdateModel.cs

```diff
@@ -17,31 +17,35 @@ internal partial class Session
             internal struct SignatureHelpSelection
             {
                 private readonly SignatureHelpItem _selectedItem;
+                private readonly bool _userSelected;
                 private readonly int? _selectedParameter;
 
-                public SignatureHelpSelection(SignatureHelpItem selectedItem, int? selectedParameter) : this()
+                public SignatureHelpSelection(SignatureHelpItem selectedItem, bool userSelected, int? selectedParameter) : this()
                 {
                     _selectedItem = selectedItem;
+                    _userSelected = userSelected;
                     _selectedParameter = selectedParameter;
                 }
 
                 public int? SelectedParameter => _selectedParameter;
                 public SignatureHelpItem SelectedItem => _selectedItem;
+                public bool UserSelected => _userSelected;
             }
 
             internal static class DefaultSignatureHelpSelector
             {
                 public static SignatureHelpSelection GetSelection(
                     IList<SignatureHelpItem> items,
                     SignatureHelpItem selectedItem,
+                    bool userSelected,
                     int argumentIndex,
                     int argumentCount,
                     string argumentName,
                     bool isCaseSensitive)
                 {
-                    var bestItem = GetBestItem(selectedItem, items, argumentIndex, argumentCount, argumentName, isCaseSensitive);
-                    var selectedParameter = GetSelectedParameter(bestItem, argumentIndex, argumentName, isCaseSensitive);
-                    return new SignatureHelpSelection(bestItem, selectedParameter);
+                    SelectBestItem(ref selectedItem, ref userSelected, items, argumentIndex, argumentCount, argumentName, isCaseSensitive);
```

Comment:
i find the 'ref' ugly :)    your call whether you want to return a tuple instead.

Reply:
Ended up leaving as `ref`